spark/src/main/scala/ai/chronon/spark/JoinUtils.scala

```diff
@@ -111,7 +111,7 @@ object JoinUtils {
     lazy val defaultLeftStart = Option(leftSource.query.startPartition)
       .getOrElse(tableUtils.firstAvailablePartition(leftSource.table, leftSource.subPartitionFilters).get)
     val leftStart = overrideStart.getOrElse(defaultLeftStart)
-    val leftEnd = Option(leftSource.query.endPartition).getOrElse(endPartition)
+    val leftEnd = Seq(Option(leftSource.query.endPartition), Some(endPartition)).flatten.min
```

Comment:
Why is `endPartition` treated differently from `overrideStartPartition`? Seems like the intention of both is to override the dates from leftSource? Seems a bit confusing to first timers

Reply:
For the right boundary of the backfill range, we still want to use end partition even if it is earlier than the query.endPartition.

Reply:
`leftSource.query.endPartition` has a different purpose compared with `overrideStartPartition`. The former indicates the latest valid ds partition of the table (after which the table may have no data); the latter is designed to override the backfill range.

Reply:
Having a earlier `leftEnd` does not violate the configuration - `leftSource.query.endPartition`